internal/cache/util/util_test.go

```diff
@@ -233,3 +234,37 @@ func (ut *utilTest) Test_IsCacheHandleValid_False() {
 		ExpectFalse(IsCacheHandleInvalid(errors.New(errMsg)))
 	}
 }
+
+func TestCheckPermissionsForShouldNotReturnAnyErrorWhenDirectoryExists(t *testing.T) {
+	base := "./" + string(testutil.GenerateRandomBytes(4))
+	dirPath := base + "/" + "path/cachedir"
+	dirCreationErr := os.MkdirAll(dirPath, 0700)
+	defer os.RemoveAll(base)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	ExpectEq(nil, dirCreationErr)
+	ExpectEq(nil, err)
+}
+
+func TestCheckPermissionsForShouldNotReturnAnyErrorWhenDirectoryCanBeCreated(t *testing.T) {
+	base := "./" + string(testutil.GenerateRandomBytes(4))
+	dirPath := base + "/" + "path/cachedir"
+	defer os.RemoveAll(base)
+
+	err := CreateCacheDirectoryIfNotPresentAt(dirPath)
+
+	ExpectEq(nil, err)
```

Comment:
we should also check if the directory is created and has expected dir permission bits set ?

Reply:
done